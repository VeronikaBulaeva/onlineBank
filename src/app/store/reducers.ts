import { combineReducers } from "@reduxjs/toolkit";
import creditSlice from "./slices/creditSlice.ts";

const rootReducer = combineReducers({
  credit: creditSlice,
});

export default rootReducer;
