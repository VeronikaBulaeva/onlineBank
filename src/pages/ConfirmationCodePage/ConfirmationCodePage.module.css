.wrapper {
    padding-top: 104px;
    padding-bottom: 96px;
    margin-inline: auto;
    max-width: 1300px;
}
