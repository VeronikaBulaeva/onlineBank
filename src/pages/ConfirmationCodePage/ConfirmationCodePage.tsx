import { FC } from "react";
import styles from "./ConfirmationCodePage.module.css";
import CodeConfirmation from "@/components/CodeConfirmation/CodeConfirmation.tsx";
import useStep from "@/lib/useStep.ts";
import { useParams } from "react-router-dom";

const STEP = 5;

const ConfirmationCodePage: FC = () => {
  const { id } = useParams<{ id: string }>();

  useStep(STEP, id);

  return (
    <div className={styles.wrapper}>
      <CodeConfirmation />
    </div>
  );
};

export default ConfirmationCodePage;
