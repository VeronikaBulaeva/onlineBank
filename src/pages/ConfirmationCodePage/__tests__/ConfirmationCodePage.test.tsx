import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import ConfirmationCodePage from "../ConfirmationCodePage.tsx";
import { Provider } from "react-redux";
import { store } from "@/app/store/store.ts";
import { BrowserRouter as Router } from "react-router-dom";

describe("ConfirmationCodePage", () => {
  it("Отрисовка компонента", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <ConfirmationCodePage />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
