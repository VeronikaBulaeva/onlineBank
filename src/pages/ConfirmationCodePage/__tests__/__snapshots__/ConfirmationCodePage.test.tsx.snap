// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfirmationCodePage > Отрисовка компонента 1`] = `
<div>
  <div
    class="_wrapper_81e6f5"
  >
    <div
      class="_grid_a6bf51"
    >
      <p
        class="_title_a6bf51"
      >
        Please enter confirmation code
      </p>
      <div
        class="_input__block_a6bf51"
      >
        <input
          class="_invisible_a6bf51"
          data-testid="codeInput"
          type="text"
        />
        <p
          class="_input_a6bf51"
          data-testid="clickText0"
        >
          <img
            alt="ellipse"
            src="/src/assets/ellipse.svg"
          />
        </p>
        <p
          class="_input_a6bf51"
          data-testid="clickText1"
        >
          <img
            alt="ellipse"
            src="/src/assets/ellipse.svg"
          />
        </p>
        <p
          class="_input_a6bf51"
          data-testid="clickText2"
        >
          <img
            alt="ellipse"
            src="/src/assets/ellipse.svg"
          />
        </p>
        <p
          class="_input_a6bf51"
          data-testid="clickText3"
        >
          <img
            alt="ellipse"
            src="/src/assets/ellipse.svg"
          />
        </p>
      </div>
    </div>
  </div>
</div>
`;
